cairo1-run/src/main.rs

```diff
@@ -290,20 +303,14 @@ fn run(args: impl Iterator<Item = String>) -> Result<Vec<MaybeRelocatable>, Erro
 
     let mut runner = CairoRunner::new_v2(&program, &args.layout, runner_mode)?;
 
-    let mut vm = VirtualMachine::new(args.trace_file.is_some());
+    let mut vm = VirtualMachine::new(args.trace_file.is_some() || args.air_public_input.is_some());
     let end = runner.initialize(&mut vm)?;
 
     additional_initialization(&mut vm, data_len)?;
 
     // Run it until the end/ infinite loop in proof_mode
     runner.run_until_pc(end, &mut vm, &mut hint_processor)?;
-
-    if args.proof_mode {
```

Comment:
why did you remove this?

Reply:
The previous code called `runner.run_until_next_power_of_2(&mut vm, &mut hint_processor)` if proof_mode was activated. This same function is called in the `end_run` method if proof_mode is activated